### PR TITLE
Add L0_batcher and L0_infer cuda shared memory tests

### PR DESCRIPTION
**Pull request by CoderHam**
Add test coverage for cuda shared memory - L0_infer and L0_batcher variants

Currently seeing test failures for String tensors in L0_infer for cuda shared memory

### CODE REVIEW

File: qa/L0_batcher/batcher_test.py [hunk 1]
```diff
@@ -39,32 +39,38 @@
...
-            shm_tmp_handle = shm.create_shared_memory_region(shm_region +'_data', '/'+ shm_region, 512)
+            if TEST_SYSTEM_SHARED_MEMORY:
+                shm_tmp_handle = shm.create_shared_memory_region(shm_region +'_data', '/'+ shm_region, 512)
+                shared_memory_ctx.register(shm_tmp_handle)
+            else:
+                shm_tmp_handle = cudashm.create_shared_memory_region(shm_region +'_data', 512, 0)
+                shared_memory_ctx.cuda_register(shm_tmp_handle)
             precreated_shm_regions.append(shm_tmp_handle)
-            shared_memory_ctx.register(shm_tmp_handle)
         return precreated_shm_regions
```
[the remark this answers is not in the document]
Comment (CoderHam) [reply qa/L0_batcher/batcher_test.py]: Done

[the remark this answers is not in the document]
Comment (CoderHam) [reply qa/L0_batcher/batcher_test.py]: Done

File: qa/L0_batcher/batcher_test.py [hunk 2]
```diff
@@ -39,32 +39,38 @@
...
 
 def _cleanup_after(shm_handles):
-    if TEST_SHARED_MEMORY:
+    if TEST_SYSTEM_SHARED_MEMORY or TEST_CUDA_SHARED_MEMORY:
```
[the remark this answers is not in the document]
Comment (CoderHam) [reply qa/L0_batcher/batcher_test.py]: Done

File: qa/common/infer_util.py [hunk 3]
```diff
@@ -86,165 +84,100 @@ def infer_exact(tester, pf, tensor_shape, batch_size,
...
+    if output1_dtype == np.object:
+        expected1_list_tmp = _prepend_string_size(expected1_list)
+    else:
+        expected1_list_tmp = expected1_list
+    output1_byte_size = sum([e1.nbytes for e1 in expected1_list])
+
+    if TEST_CUDA_SHARED_MEMORY or TEST_SYSTEM_SHARED_MEMORY:
+        shm_handles = su.create_register_set_shm_regions(input0_list, input1_list, expected0_list_tmp, \
+                                        expected1_list_tmp, outputs, shm_region_names, precreated_shm_regions)
     for config in configs:
```
[the remark this answers is not in the document]
Comment (CoderHam) [reply qa/common/infer_util.py]: Done

File: qa/L0_sequence_batcher/sequence_batcher_test.py [hunk 4]
```diff
@@ -47,9 +47,7 @@
 else:
   import Queue as queue
 
-_test_shared_memory = False
-if "TEST_SHARED_MEMORY" in os.environ:
-  _test_shared_memory = True if (os.environ["TEST_SHARED_MEMORY"] == "1") else False
+TEST_SYSTEM_SHARED_MEMORY = bool(int(os.environ.get('TEST_SYSTEM_SHARED_MEMORY', 0)))
```
[the remark this answers is not in the document]
Comment (CoderHam) [reply qa/L0_sequence_batcher/sequence_batcher_test.py]: Done

File: qa/L0_batcher/batcher_test.py [hunk 5]
```diff
@@ -39,35 +39,42 @@
...
         return precreated_shm_regions
+    return []
 
 def _cleanup_after(shm_handles):
-    if TEST_SHARED_MEMORY:
+    if len(shm_handles) != 0:
         shared_memory_ctx = SharedMemoryControlContext("localhost:8000", ProtocolType.HTTP, verbose=True)
-        for shm_tmp_handle in shm_handles:
-            shared_memory_ctx.unregister(shm_tmp_handle)
+    for shm_tmp_handle in shm_handles:
```
Comment (deadeyegoodwin) [inline qa/L0_batcher/batcher_test.py]: These lines need to be indented...

Comment (CoderHam) [reply qa/L0_batcher/batcher_test.py]: Fixed

**Pull request merged**